runtime/near-vm-runner/src/wasmer_runner.rs

```diff
@@ -114,8 +113,11 @@ impl IntoVMError for wasmer_runtime::error::RuntimeError {
                 }
                 // A trap that Wasmer knows about occurred.
                 // As of 0.17.1, can be thrown on C signals caught, for example OOM.
-                InvokeError::TrapCode { code: _, srcloc: _ } => {
-                    VMError::FunctionCallError(WasmUnknownError)
+                InvokeError::TrapCode { code, srcloc } => {
+                    VMError::FunctionCallError(FunctionCallError::Nondeterministic(format!(
+                        "Wasmer trap {} at {}",
+                        *code as u32, srcloc
+                    )))
                 }
```

Comment:
Hm, from the light reading of the code, I am not sure that this should *always* be nondeterministic. It seems that, eg, division by zero generates this trap, no? Do we have a test for division by zero?

Reply:
Correct, reworked that piece.